bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,252 @@
+                        : new DateTimeType(pet.dateOfBirth.atStartOfDay(ZoneId.systemDefault())));
+                updateState(PET_CHANNEL_WEIGHT,
+                        pet.weight == null ? UnDefType.UNDEF : new QuantityType<Mass>(pet.weight, SIUnits.KILOGRAM));
+                if (pet.tagId != null) {
+                    SurePetcareTag tag = petcareAPI.getTag(pet.tagId.toString());
+                    if (tag != null) {
+                        updateState(PET_CHANNEL_TAG_IDENTIFIER, new StringType(tag.tag));
+                    }
+                }
+                if (pet.status.activity.deviceId != null) {
+                    SurePetcareDevice device = petcareAPI.getDevice(pet.status.activity.deviceId.toString());
+                    if (device != null) {
+                        updateState(PET_CHANNEL_LOCATION_CHANGED_THROUGH, new StringType(device.name));
+                    }
+                } else if (pet.status.activity.userId != null) {
+                    SurePetcareHousehold household = petcareAPI.getHousehold(pet.householdId.toString());
+                    if (household != null) {
+                        Long userId = pet.status.activity.userId;
+                        household.users.stream().map(user -> user.user).filter(user -> userId.equals(user.userId))
+                                .forEach(user -> updateState(PET_CHANNEL_LOCATION_CHANGED_THROUGH,
+                                        new StringType(user.userName)));
+                    }
+                }
+                SurePetcarePetFeeding feeding = pet.status.feeding;
+                if (feeding != null) {
+                    SurePetcareDevice device = petcareAPI.getDevice(feeding.deviceId.toString());
+                    if (device != null) {
+                        updateState(PET_CHANNEL_FEEDER_DEVICE, new StringType(device.name));
+                        int bowlId = device.control.bowls.bowlId;
+                        int numBowls = feeding.feedChange.size();
+                        for (int i = 0; (i < 2) && (i < numBowls); i++) {
+                            if (bowlId == 1) {
+                                updateState(PET_CHANNEL_FEEDER_LAST_CHANGE,
+                                        new QuantityType<Mass>(feeding.feedChange.get(i), SIUnits.GRAM));
+                            } else if (bowlId == 4) {
+                                if ((i + 1) == 1) {
+                                    updateState(PET_CHANNEL_FEEDER_LAST_CHANGE_LEFT,
+                                            new QuantityType<Mass>(feeding.feedChange.get(i), SIUnits.GRAM));
+                                }
+                                if ((i + 1) == 2) {
+                                    updateState(PET_CHANNEL_FEEDER_LAST_CHANGE_RIGHT,
+                                            new QuantityType<Mass>(feeding.feedChange.get(i), SIUnits.GRAM));
+                                }
+                            }
+                        }
```

Comment:
So my understanding so far, is that there are two different types of feeder bowls, one with a single bowl, and another with two bowls side by side.

If that is the case then you shouldn't have your logic in a loop at all.
```suggestion 
                        if(numBowls > 0){
                            if (bowlId == 1) {
                                updateState(PET_CHANNEL_FEEDER_LAST_CHANGE,
                                        new QuantityType<Mass>(feeding.feedChange.get(0), SIUnits.GRAM));
                            } else if (bowlId == 4) {
                                updateState(PET_CHANNEL_FEEDER_LAST_CHANGE_LEFT,
                                            new QuantityType<Mass>(feeding.feedChange.get(0), SIUnits.GRAM));
                                if (numBowls > 1) {
                                    updateState(PET_CHANNEL_FEEDER_LAST_CHANGE_RIGHT,
                                            new QuantityType<Mass>(feeding.feedChange.get(1), SIUnits.GRAM));
                                }
                            }
                        }
                        
```

Please update the logic in the SurePetcareDeviceHandler similarly.

Reply:
@HerzScheisse : Is there also a case where the setting is "2 small bowls", but only one is used (i.e. the API returns only one feedChange record)? If not, I'll make the change as suggested by @cpmeister .

Reply:
Change implemented as suggested.